### ISSUE TITLE
Plugin parameters are not initialized properly on some platforms

### ISSUE THREAD

**Issue by michaelwillis** [michaelwillis/dragonfly-reverb#88]
This code is broken on some platforms:

```
  for (uint32_t param = 0; param < paramCount; param++) {
    newParams[param] = banks[DEFAULT_BANK].presets[DEFAULT_PRESET].params[param];
    oldParams[param] = FP_NAN;
  }
```

It was written with the assumption that FP_NAN does not equal any number that can be set on the the dials or sliders, but on some platforms, FP_NAN equals zero, which results in parameters not getting initialized.

**Comment by michaelwillis** [comment]
@colin-m-fletcher shed some light on the subject in [a comment elsewhere](https://github.com/michaelwillis/dragonfly-reverb/pull/86):

> FP_NAN isn't a NaN itself: it's an enumerated return value from fpclassify(x) indicating that x is NaN. To get an actual NaN constant value, you need nan("") (or std::nan("") in C++).

**Comment by michaelwillis** [comment]
Resolved by 554e1ad2e54168a75823daa6b6d1a55a2a318fff
